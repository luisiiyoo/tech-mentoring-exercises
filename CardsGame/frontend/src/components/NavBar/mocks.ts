import { NavItem } from '../../model/navItem';

export const navItemMock: NavItem[] = [
  {
    route: 'my-home',
    title: 'My Home',
    iconClass: 'fa fa-fw fa-heartbeat',
  },
  {
    route: 'free-code-camp',
    title: 'Free Code Camp',
    iconClass: 'fa fa-fw fa-beer',
    childs: [
      {
        idChild: '1',
        route: 'sub-page',
        title: 'SubPage',
      },
    ],
  },
];

export const stateMock = {
  main: {
    route: 'home',
    expand: true,
  },
};
