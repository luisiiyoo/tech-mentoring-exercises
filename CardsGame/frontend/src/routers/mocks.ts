import { NavItem } from '../model/navItem';

export const navBarItemsMock: NavItem[] = [
  {
    route: 'home',
    title: 'Home',
    iconClass: 'fa fa-fw fa-home',
  },
  {
    route: 'engineering',
    title: 'Engineering',
    iconClass: 'fa fa-fw fa-graduation-cap',
    childs: [
      {
        idChild: '1',
        route: 'basic-back-end',
        title: 'Basic Back-End',
      },
      {
        idChild: '2',
        route: 'avanced-back-end',
        title: 'Avanced Back-End',
      },
      {
        idChild: '3',
        route: 'basic-front-end',
        title: 'Basic Front-End',
      },
      {
        idChild: '4',
        route: 'avanced-front-end',
        title: 'Avanced Front-End',
      },
      {
        idChild: '5',
        route: 'qa',
        title: 'Q&A',
      },
    ],
  },
];

export const stateMock = {
  main: {
    expand: true,
  },
  topicDetails: {
    visible: false,
    topic: {
      id: -1,
      name: '',
      level: -1,
      shortDescription: '',
      description: '',
      type: '',
      tags: [],
      metadata: {},
      resources: [
        {
          id: -1,
          name: '',
          shortDescription: '',
          description: '',
          url: '',
          complexity: '',
          tags: [],
          metadata: {},
        },
      ],
    },
  },
};
