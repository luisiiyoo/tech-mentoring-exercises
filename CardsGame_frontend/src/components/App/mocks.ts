export const navItemsMock = [
  {
    route: 'my-home',
    title: 'My Home',
    iconClass: 'fa fa-fw fa-heartbeat',
  },
  {
    route: 'free-code-camp',
    title: 'Free Code Camp',
    iconClass: 'fa fa-fw fa-beer',
    childs: [
      {
        idChild: '1',
        route: 'sub-page',
        title: 'SubPage',
      },
    ],
  },
];
