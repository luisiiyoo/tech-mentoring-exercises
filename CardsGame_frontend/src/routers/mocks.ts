import { NavItem } from '../model/navItem';

export const navBarItemsMock: NavItem[] = [
  {
    route: 'home',
    title: 'Home',
    iconClass: 'fa fa-fw fa-home',
  },
  {
    route: 'engineering',
    title: 'Engineering',
    iconClass: 'fa fa-fw fa-graduation-cap',
    childs: [
      {
        journeyID: '1',
        route: 'basic-back-end',
        title: 'Basic Back-End',
      },
      {
        journeyID: '2',
        route: 'avanced-back-end',
        title: 'Avanced Back-End',
      },
      {
        journeyID: '3',
        route: 'basic-front-end',
        title: 'Basic Front-End',
      },
      {
        journeyID: '4',
        route: 'avanced-front-end',
        title: 'Avanced Front-End',
      },
      {
        journeyID: '5',
        route: 'qa',
        title: 'Q&A',
      },
    ],
  },
];

export const stateMock = {
  main: {
    expand: true,
  },
  topicDetails: {
    visible: false,
    topic: {
      id: -1,
      name: '',
      level: -1,
      shortDescription: '',
      description: '',
      type: '',
      tags: [],
      metadata: {},
      resources: [
        {
          id: -1,
          name: '',
          shortDescription: '',
          description: '',
          url: '',
          complexity: '',
          tags: [],
          metadata: {},
        },
      ],
    },
  },
};
