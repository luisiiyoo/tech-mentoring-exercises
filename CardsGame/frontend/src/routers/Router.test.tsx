import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import { useSelector } from 'react-redux';
import { stateMock, navBarItemsMock } from './mocks';
import Routes from './Router';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}));

describe('JourneyRouter', () => {
  beforeEach(() => {
    useSelector.mockImplementation((callback) => callback(stateMock));
  });

  afterEach(() => {
    cleanup();
    useSelector.mockClear();
  });

  it('should take a snapshot', () => {
    const navBarItemsMock = [
      {
        route: 'home',
        title: 'Home',
        iconClass: 'fa fa-fw fa-home',
      },
    ];
    const { asFragment } = render(<Routes navBarItems={navBarItemsMock} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the root page', () => {
    const { getByTestId } = render(<Routes navBarItems={navBarItemsMock} />);
    expect(getByTestId('mainComponent').innerHTML).toMatch('Cards Game');
  });

  it('should render the home page', async () => {
    const { getByTestId } = render(<Routes navBarItems={navBarItemsMock} />);
    fireEvent.click(getByTestId('home'));
    expect(getByTestId('mainComponent')).toHaveTextContent(`Cards Game`);
  });

  it('should render 404 Not Found message when an invalid route or not provided component is reached', () => {
    const { getByTestId } = render(
      <Routes
        navBarItems={[
          {
            route: 'route-mock',
            title: 'Route mock without a component defined',
          },
        ]}
      />,
    );
    fireEvent.click(getByTestId('route-mock'));
    expect(getByTestId('ErrorMessage').innerHTML).toMatch('Page not found');
  });
});
