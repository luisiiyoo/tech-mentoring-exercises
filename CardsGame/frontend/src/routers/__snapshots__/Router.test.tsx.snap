// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JourneyRouter should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="sidenav---sidenav---_2tBP sidenav---expanded---1KdUL"
  >
    <button
      aria-expanded="true"
      class="sidenav---sidenav-toggle---1KRjR"
      data-testid="toggleNavBar"
      role="button"
    >
      <span
        class="sidenav---icon-bar---u1f02"
      />
      <span
        class="sidenav---icon-bar---u1f02"
      />
      <span
        class="sidenav---icon-bar---u1f02"
      />
    </button>
    <div
      class="NavBarHeader"
      data-testid="headerContainer"
      style="display: block;"
    >
      <div
        class="NavBarHeader-Title"
        data-testid="headerTitle"
      >
        Cards Game
      </div>
    </div>
    <div
      class="sidenav---sidenav-nav---3tvij sidenav---expanded---1KdUL"
      role="menu"
    >
      <div
        class="sidenav---sidenav-navitem---uwIJ-"
        role="presentation"
      >
        <div
          class="sidenav---navitem---9uL5T"
          data-testid="home"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="sidenav---navicon---3gCRo"
          >
            <i
              class="fa fa-fw fa-home NavBar-Icon"
            />
          </div>
          <div
            class="sidenav---navtext---1AE_f"
            style="font-size: 1.3em; vertical-align: middle;"
          >
            Home
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="Container"
  >
    <div
      class="MainComponent"
      data-testid="mainComponent"
      style="margin-left: 240px;"
    >
      <div
        class="HomePage"
      >
        <h2>
          Cards Game App
        </h2>
        <h3>
          App Description
        </h3>
        <h4>
          Instructions
        </h4>
        <p>
          Instructions will be here.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
